source/state_representation/test/tests/space/cartesian/test_cartesian_pose.cpp

```diff
@@ -5,7 +5,6 @@
 using namespace state_representation;
 
 static void expect_only_pose(CartesianPose& pose) {
-  EXPECT_EQ(static_cast<CartesianState&>(pose).get_orientation().norm(), 1);
```

Comment:
How come this test was not failing before? Can you actually put a less strict version with almost equal. This condition should still be true.

Reply:
Well, first I did replace it with 
`EXPECT_NEAR(static_cast<CartesianState&>(pose).get_orientation().norm(), 1.0, 1e-9);`, which passed. But then I realised that this condition is not something that the unit test is actually supposed to check. The purpose of this helper is to ensure the other state variables are precisely zero. You can always add more checks, but it starts to defeat the point of unit testing.

Why did it pass earlier and not now? My best assertion is that our "random" tests are actually deterministic to a certain extent. For a given test structure, the full test evaluation will always be the same. But, if you change the structure of the tests like I've done here with the folders, gtest will execute them in a different order, and the random seed will be different by the time it reaches this test case. This lead to a random orientation which had a floating point imprecision when calculating the norm (though we are really talking about the single mantissa bit or so, making it 1.00000000000001 or whatever). Long story short, we should never use `EQ` on floating point numbers unless they are true 0, and we should not test assertions that are not part of a unit case.

Reply:
ok yes makes sense. Thanks